db/collection.go

```diff
@@ -954,6 +956,10 @@ func (c *collection) save(
 				return cid.Undef, client.NewErrFieldNotExist(k)
 			}
 
+			if fieldDescription.Typ != client.LWW_REGISTER {
```

Comment:
It's because by default it's LWW so we only need to update the type on the `val` variable if it's not LWW. I changed it to be set in all cases to keep in simple.